### PR TITLE
"Ft milestone5: Adding state to App and Event Handlers"

### PR DESCRIPTION
**Pull request by didymus707**
Here:- 
- state was added to the App 
- event handlers were passed down to child components as props
- Display and Button panel were rendered in App

### CODE REVIEW

**Review by Forison — COMMENTED** [review]
# Invalid❗️
#### Make sure to write a detailed pull request especially when you have created  P/R. It is not good practice to recreate an entirely new pull request after a review this makes tracking previous review difficult, could lead to inconsistentcies and time wasting in reviewing.
Kindly comment well the pull request so we can know the evolution of your work or revert to those PR.
Regards
Addo Forison

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__I must commend your efforts so far in this milestone, and the project as a whole. Here are the major highlights for me:__

- All linter checks passed successfully :100: 
- Your Readme looks good :+1: 

__However, some changes are required before this milestone can be approved. Please check my comments below and fix them as appropriate.__

**[ REQUIRED CHANGES ]**

_**Looks like there's an issue with this PR. Having checked your code, I cannot find any implementation of milestone-5 features. Kindly check if you submitted the right branch. If the submission is right, then kindly make the required changes below.**_

_**Also note that I could not find the previous reviews on this PR, whereas it showed that this project has been reviewed 3 times.**_

- [ ] It is not enough for your App component to have states: it is required of you to convert the App component to a stateful component using class, and not a functional component. ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone5/app/stateful.png))
- [ ] The states of the App component should be initialized in the constructor of the class ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone5/app/stateful.png)).
- [ ] The App component should also have `handleClick` implemented in the constructor. 
`i.e. this.handleClick = this.handleClick.bind(this);`
- [ ] The `handleClick` should be passed as `clickHandler` property to `ButtonPanel`. 
i.e. `<ButtonPanel clickHandler={this.handleClick} />`
- [ ] Also, in the App component, the `total` or `next` value should be passed from the state to `Display`. 
**The `BottonPanel` component**
- [ ]  It should implement the `handleClick` function. 
```
handleClick = (buttonName) => {
  this.props.clickHandler(buttonName);
};
```
- [ ] it also passes handleClick as `clickHandler` props to each Button ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone5/panel/passToButton.png)).
**The `Button` component**
- [ ]  it attaches `handleClick` to the `onClick` event on Button


* * *
**Please make the required changes and submit another review request when you are done.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Pull request merged**
